joomla/joomla-cms
# Fixes installer leftovers in the Joomla tmp folder after unsuccessful installation with wrong file type

**This PR includes already https://github.com/joomla/joomla-cms/pull/2143 so should be MERGED AFTER https://github.com/joomla/joomla-cms/pull/2143 .**

Related tracker issue item: http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=32207

To test:
1. Check that your joomla tmp folder is empty
2. Try insatlling a pdf file with all 3 existing methods and verify that they are not installed, and see that there are files leftover in tmp
3. Empty your tmp folder
4. Apply patch
5. Try insatlling a pdf file with all 3 methods and verify that they are not installed, but also that now with the patch there are no leftovers in the tmp folder.
6. install an extension by all 4 methods and verify for each that it is still installed correctly, and that there are still no leftovers in tmp

BTW: this is NOT a vulnerability or security issue, as only super-admins have installation rights, and with those rights, they can store files already in the media manager, or install components that allow file access.
